testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +112,69 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        assert D._check_set_unit(units) == None
+
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        D._check_set_unit(units)
+        del D.units['density']
+        D._check_set_unit({})
+        assert D.units['density'] == None
+
+    def test_parameters_isdensity(self, D):
+        with pytest.warns(UserWarning, match='Running make_density()'):
+            D.make_density()
+
+    def test_check_convert_density_grid_not_density(self, D1):
+        unit = 'nm^{-3}'
+        with pytest.raises(RuntimeError, match="The grid is not a density"):
+            D1.convert_density(unit)
+
+    def test_check_convert_density_value_error(self, D):
+        unit = 'A^{-2}'
+        with pytest.raises(ValueError, match="The name of the unit"):
+            D.convert_density(unit)
+
+    def test_check_convert_density_units_same_density_units(self, D):
+        unit = 'A^{-3}'
+        D.convert_density(unit)
+        assert unit == D.units['density']
+
+    def test_check_convert_density_units_density(self, D):
+        unit = 'nm^{-3}'
+        D.convert_density(unit)
+        assert D.units['density'] == unit
+
+    def test_convert_length_same_length_units(self, D):
+        unit = 'A'
+        D.convert_length(unit)
+        assert unit == D.units['length']
```

Comment:
slightly more readable if you put the reference on the RHS
```suggestion
        assert D.units['length'] == unit
```